### PR TITLE
fix(sdk): Update syntax for lightweight components v2.

### PR DESCRIPTION
**Pull request by neuromage**
**Description of your changes:**
Use `Input[T]` and `Output[T]` for representing input/output artifacts
of type `T`. This enables users to see these annotations as merely the
classes `T`, enabling autocompletion for `T`, reducing verbosity and simplifying
I/O types code.

Syntax for defining such components looks like the following:
```py
@component
def my_component(
    # Input artifact of type `Dataset`.
    dataset: Input[Dataset],
    # Output artifact of type `Model`.
    model: Output[Model],
    # Output artifact of type `ClassificationMetrics`.
    metrics: Output[ClassificationMetrics]):
  '''My component. '''

  # Access input and output files through .path or .uri
  with open(dataset.path, 'r') as f:
    ...
 
  # Set metadata for output artifacts.
  model.metadata['framework'] = 'TensorFlow'

  # Use metrics to log
  metrics.log_roc_curve(...)
```

Other changes include:
- no longer need to use .get() to obtain the artifact class
- no auto-creation of the `data` file for output artifacts. This enables
  artifacts to be either directories or files.

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/components/_python_op.py [hunk 1]
```diff
@@ -328,15 +328,28 @@ def annotation_to_type_struct(annotation):
     input_names = set()
     output_names = set()
     for parameter in parameters:
-        parameter_annotation = parameter.annotation
+        # parameter_annotation = parameter.annotation
```
Comment (chensun) [inline sdk/python/kfp/components/_python_op.py:331]: nit: remove?

**Comment by neuromage** [comment]
/cc @amygdala 
/cc @joeliedtke 

@amygdala I think this fixes the issue with directory creation as well. I removed the implicit creation of `data` files here.

File: sdk/python/kfp/components/_python_op.py (hunk 1 above)
Comment (neuromage) [reply sdk/python/kfp/components/_python_op.py]: Done.

File: sdk/python/kfp/components/_python_op.py [hunk 2]
```diff
@@ -328,15 +328,31 @@ def annotation_to_type_struct(annotation):
     input_names = set()
     output_names = set()
     for parameter in parameters:
-        parameter_annotation = parameter.annotation
+        parameter_type = parameter.annotation
```
Comment (Ark-kun) [inline sdk/python/kfp/components/_python_op.py:331]: I wonder what is the main motivation for this change?
Not all annotations are types. In fact most KFP annotations are not types, but are strings (type names).

**Pull request merged**